### PR TITLE
New metric schema

### PR DESCRIPTION
**Pull request by mkleina**
Plugin updated to new metric schema, now compatible with snap master (v0.13.0-beta-73-g170c8d3)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,8 +3,8 @@ services:
 - mysql
 language: go
 go:
-- 1.4.3
 - 1.5.3
+- 1.6
```
Comment (candysmurf) [inline .travis.yml]: we support both go1.5.4 and go1.6.2

File: mysql/mysql.go [hunk 2]
```diff
@@ -104,13 +104,13 @@ func (s *mysqlPublisher) Publish(contentType string, content []byte, config map[
...
 	for _, m := range metrics {
-		key = sliceToString(m.Namespace())
+		key = sliceToString(m.Namespace().Strings())
 		value, err = interfaceToString(m.Data())
 		if err != nil {
 			logger.Printf("Error: %v", err)
 			return err
 		}
-		_, err := insert.Exec(m.Timestamp(), m.Source(), key, value)
+		_, err := insert.Exec(m.Timestamp(), m.Tags()["source"], key, value)
```
Comment (candysmurf) [inline mysql/mysql.go]: probably using core.STD_TAG_PLUGIN_RUNNING_ON representing "source" or a constant is better

File: mysql/mysql_integration_test.go [hunk 3]
```diff
@@ -28,18 +28,19 @@ import (
...
 )
 
 func TestMySQLPublish(t *testing.T) {
 	var buf bytes.Buffer
-	metrics := []plugin.PluginMetricType{
-		*plugin.NewPluginMetricType([]string{"test", "string"}, time.Now(), "127.0.0.1", nil, nil, "example_string"),
-		*plugin.NewPluginMetricType([]string{"test", "int"}, time.Now(), "127.0.0.1", nil, nil, 1),
-		*plugin.NewPluginMetricType([]string{"test", "string", "slice"}, time.Now(), "localhost", nil, nil, []string{"str1", "str2"}),
-		*plugin.NewPluginMetricType([]string{"test", "string", "slice"}, time.Now(), "localhost", nil, nil, []int{1, 2}),
+	metrics := []plugin.MetricType{
```
Comment (candysmurf) [inline mysql/mysql_integration_test.go]: same comment for the "source" as above

**Comment by candysmurf** [comment]
LGTM if you address those comments.

File: mysql/mysql_integration_test.go (hunk 3 above)
Comment (mkleina) [inline mysql/mysql_integration_test.go]: Ok, fixed

**Comment by candysmurf** [comment]
👍

**Pull request merged**